app/components/questions.js

```diff
@@ -0,0 +1,108 @@
+// https://github.com/EnCiv/undebate-ssp/issues/48
+import React, { useState, useEffect } from 'react'
+import { createUseStyles } from 'react-jss'
+import CountLimitedTextInput from './count-limited-text-input'
+import Submit from './submit'
+
+export default function Questions({ electionOM, onDone }) {
+    const classes = useStyles()
+    const { electionMethods } = electionOM
+    const [questions, setQuestions] = useState([''])
+    const [error, setError] = useState('')
+    const [isValid, setIsValid] = useState(false)
+
+    const checkValid = () =>
+        !electionMethods.areQuestionsLocked() &&
+        (questions.length > 1 || (questions.length === 1 && !questions.includes('')))
+
+    useEffect(() => {
+        setIsValid(checkValid())
+    }, [isValid, electionMethods, questions])
+
+    useEffect(() => {
+        if (!questions.includes('')) {
+            setError('')
+        }
+    }, [questions, error])
+
+    const addQuestion = () => {
+        if (!questions.includes('')) {
+            setError('')
+            setQuestions([...questions, ''])
+        } else {
+            setError('All questions must not be empty!')
+        }
+    }
+
+    const generateKey = pre => {
+        return `${pre}_${new Date().getTime()}`
+    }
+
+    return (
+        <div className={classes.container}>
+            <div className={classes.send}>
+                <span>What questions would you like to ask the candidates?</span>
+                <Submit
+                    disabled={!isValid}
+                    disableOnClick
+                    onDone={() => {
+                        onDone({
+                            value: questions.filter(question => question !== ''),
+                            valid: checkValid(),
+                        })
+                    }}
+                />
+            </div>
+            <div className={classes.questionBox}>
+                {questions.map((question, index) => (
+                    <CountLimitedTextInput
+                        key={generateKey(index)}
+                        name={`Question ${index + 1}`}
+                        maxCount={250}
+                        defaultValue={question}
+                        onDone={props => {
+                            setQuestions([...questions.slice(0, index), props.value, ...questions.slice(index + 1)])
```

Comment:
Here is a bit more of a challenge.  If props.valid, Call electionMethods.upsert({questions: {[key]: props.value}) this is how we are going to update the database. You can still do setQuestions or, you might not need to because the thing you are really keeping track of in this component is whether the question has an answer or not - this module doesn't really care about the question's content.  And in the story, you will need to create a dummy upsert. For now, you could just show on screen the result of the upsert like you show the result of the onDone.

Reply:
If we use if valid, I see one problem: when the user change one of the question on the top to empty, then that question is not valid (length must > 0 to be valid)
The point is when user click on add question button, then the text area will show, but what happen if the user only want 2 questions only, but the text area is already 5. For example, I think user can fill in number 3, 4 and leave text area be empty in number 1,2 and 5
So I just setQuestions and call electionMethods.upsert() without checking props.valid